src/tree/registries/RegistriesTreeItem.ts

```diff
@@ -3,72 +3,188 @@
+                    const provider = getRegistryProviders().find(rp => rp.id === cachedProvider.id);
+                    if (!provider) {
+                        throw new Error(`Failed to find registry provider with id "${cachedProvider.id}".`);
+                    }
 
-        const azureAccountTreeItem = await this._azureAccountTreeItemTask;
-        if (azureAccountTreeItem) {
-            result.push(azureAccountTreeItem);
-        }
+                    const parent = provider.isSingleRegistry ? this._connectedRegistriesTreeItem : this;
+                    const treeItem = this.initTreeItem(new provider.treeItemType(parent, cachedProvider));
+                    if (provider.isSingleRegistry) {
+                        this._connectedRegistriesTreeItem.children.push(treeItem);
+                        return undefined;
+                    } else {
+                        return treeItem;
+                    }
+                },
+                cachedInfo => cachedInfo.id
+            );
 
-        result.push(this._privateRegistriesTreeItem);
+            if (this._connectedRegistriesTreeItem.children.length > 0) {
+                children.push(this._connectedRegistriesTreeItem);
+            }
 
-        return result;
+            return children;
+        }
     }
 
     public hasMoreChildrenImpl(): boolean {
         return false;
     }
 
-    public compareChildrenImpl(): number {
-        return 0; // items already sorted
-    }
+    public async connectRegistry(context: IActionContext): Promise<void> {
+        let picks: IAzureQuickPickItem<IRegistryProvider | undefined>[] = getRegistryProviders().map(rp => {
+            return {
+                label: rp.label,
+                description: rp.description,
+                detail: rp.detail,
+                data: rp
+            };
+        })
+        picks = picks.sort((p1, p2) => p1.label.localeCompare(p2.label));
+        picks.push({
+            label: "$(link-external) Don't see your provider? Learn how to contribute...",
+            data: undefined
+        });
 
-    public async pickTreeItemImpl(expectedContextValues: (string | RegExp)[]): Promise<AzExtTreeItem | undefined> {
-        let matchesAzure = false;
-        let matchesOther = false;
-        for (const rt of Object.values(RegistryType)) {
-            for (const val of expectedContextValues) {
-                if (rt === RegistryType.azure) {
-                    if (isAncestoryOfRegistryType(val, <RegistryType>rt)) {
-                        matchesAzure = true;
-                        break;
-                    }
-                } else if (isAncestoryOfRegistryType(val, <RegistryType>rt)) {
-                    matchesOther = true;
-                    break;
-                }
+        let placeHolder: string = 'Select the provider for your registry';
+        const provider = (await ext.ui.showQuickPick(picks, { placeHolder, suppressPersistence: true })).data;
```

Comment:
Don't we have a learnMore property that we can leverage?

Reply:
That's only for warning messages, not quick picks. Also that logic will re-show the message, but there's no point of re-showing the quick pick if your provider isn't listed